### PR TITLE
Add CMake build system and move/rename examples

### PR DESCRIPTION
**Pull request by jopadan**

### CODE REVIEW

**Comment by Loobinex** [comment]
@jopadan I used to build adikted by typing 'make all' in WSL in adikted/mapslang. Trying to build your PR and it does not find a makefile. How can one build now?

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+add_test( putgems putgems)
+
+add_executable( puttrain ${EXAMPLES_PUTTRAIN_SOURCES} )
+target_link_libraries( puttrain PUBLIC adikted m dl)
+set_target_properties( putgems PROPERTIES RUNTIME_OUTPUT_DIRECTORY
+	${CMAKE_BINARY_DIR} )
+add_test( puttrain puttrain )
+
+add_executable( viewmap ${EXAMPLES_VIEWMAP_SOURCES} )
+target_link_libraries( viewmap PUBLIC adikted m dl ${SDL_LIBRARIES})
```
Comment (dotted) [inline CMakeLists.txt:48]: Change `${SDL_LIBRARIES}` to `${SDL_LIBRARY}` as it only exists in CMake 3.19 and onwards

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+set_target_properties( viewmap PROPERTIES RUNTIME_OUTPUT_DIRECTORY
+	${CMAKE_BINARY_DIR} )
+add_test( viewmap viewmap )
+
+add_executable( putemple ${EXAMPLES_PUTEMPLE_SOURCES} )
+target_link_libraries( putemple PRIVATE adikted m dl ${SDL_LIBRARIES})
```
Comment (dotted) [inline CMakeLists.txt:54]: Change `${SDL_LIBRARIES}` to `${SDL_LIBRARY}` as it only exists in CMake 3.19 and onwards

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+file( GLOB_RECURSE EXAMPLES_PUTGEMS_SOURCES examples/putgems/*.c
+	examples/putgems/*.h )
+file( GLOB_RECURSE EXAMPLES_PUTTRAIN_SOURCES examples/puttrain/*.c
+	examples/puttrain/*.h )
+file( GLOB_RECURSE EXAMPLES_VIEWMAP_SOURCES examples/viewmap/*.c
+	examples/viewmap/*.h )
+file( GLOB_RECURSE EXAMPLES_PUTEMPLE_SOURCES examples/putemple/*.c
+	examples/putemple/*.h )
+
+find_package( SDL REQUIRED SDL::Main )
```
Comment (dotted) [inline CMakeLists.txt:29]: Consider making it optional so if SDL is not found the `viewmap` and `putemple` example projects are not built. Not super important though.

**Comment by dotted** [comment]
Good work, just fix the `${SDL_LIBRARIES}` issue and I will be happy to merge this.

**Comment by dotted** [comment]
It would be nice if we could retain the possibility of building `mapslang` directly from within the `mapslang` folder, instead of having to build from the parent directory.

**Review by dotted — CHANGES_REQUESTED** [review]

**Comment by dotted** [comment]
Continued in #48

**Pull request closed without merging**